ONSdigital/census-rm-print-file-generator
# First version of print file generator

## Motivation and Context
We want to be able to process a sample file and generate initial print files without having to run the whole of RM. This will require modifications to the RM services and sample loader to be able to ingest the processed sample file back into RM later.

Performance wise this script takes around 70 seconds per hundred thousand sample units on my machine. This is limited by having to fetch IAC's for every sample unit.

This first version uses a placeholder uuid for `QID` and has mappings for England and Wales initial contact letters. Adding more initial contact letters with the same template should be trivial, just add the treatment country to productpack code mapping. 

The script does not yet produce a print manifest file.

## What has Changed
- Add process sample file script
- Add generate print files script
- Add unit tests

## How to Test
Run a sample file through the process sample file script, then the processed file through the generate print files script to get the print files. Check the print files match the print file spec.

## Links
https://trello.com/c/gzaskio6/647-investigate-script-to-generate-print-files-from-sample-file-directly
